### PR TITLE
Add `TabularDataset.convert/with_converter`

### PR DESCRIPTION
**Pull request by Hakuyume**
@beam2d, @okapies I renamed `fetch_post` to `convert`.

### CODE REVIEW

**Comment by Hakuyume** [comment]
jenkins, test this please

**Comment by Hakuyume** [comment]
jenkins, test this please

**Comment by Hakuyume** [comment]
jenkins, test this please

**Comment by Hakuyume** [comment]
jenkins, test this please

File: chainer/dataset/tabular/tabular_dataset.py [hunk 1]
```diff
@@ -147,6 +147,28 @@ def fetch(self):
         elif self.mode is None:
             return examples[0]
 
+    def convert(self, data):
+        """Convert fetched data.
+
+        This method takes data fetched by :meth:`fetch` and
+        pre-process them before passing them to models.
+        The default behaviour is converting each column into an ndarray.
+        This behaviour can be overloaded by :meth:`with_converter`.
```
Comment (beam2d) [inline chainer/dataset/tabular/tabular_dataset.py]:
```suggestion
        This behaviour can be overrided by :meth:`with_converter`.
```

File: chainer/dataset/tabular/tabular_dataset.py [hunk 2]
```diff
@@ -221,6 +243,21 @@ def transform_batch(self, keys, transform_batch):
         return chainer.dataset.tabular._transform._TransformBatch(
             self, keys, transform_batch)
 
+    def with_converter(self, converter):
+        """Overload the behaviour of :meth:`convert`.
```
Comment (beam2d) [inline chainer/dataset/tabular/tabular_dataset.py]:
```suggestion
        """Override the behaviour of :meth:`convert`.
```

File: chainer/dataset/tabular/tabular_dataset.py [hunk 3]
```diff
@@ -221,6 +243,21 @@ def transform_batch(self, keys, transform_batch):
...
+
+        This method overloads :meth:`convert`.
```
Comment (beam2d) [inline chainer/dataset/tabular/tabular_dataset.py]:
```suggestion
        This method overrides :meth:`convert`.
```

File: chainer/dataset/tabular/_concat.py [hunk 4]
```diff
@@ -108,3 +108,6 @@ def get_examples(self, indices, key_indices):
                     [examples[dataset_index][col_index][p]
                      for dataset_index, p in example_indices]
                     for col_index in six.moves.range(n_cols))
+
+    def convert(self, data):
+        return self._datasets[0].convert(data)
```
Comment (beam2d) [inline chainer/dataset/tabular/_concat.py]: It should be documented that the converter of the first dataset is used in `concat` and `join`. The docstring of `TabularDataset.convert` may be a good place to add a note.

**Comment by Hakuyume** [comment]
@beam2d Thank you for your comments. I updated.

File: chainer/dataset/tabular/tabular_dataset.py [hunk 5]
```diff
@@ -147,6 +147,30 @@ def fetch(self):
         elif self.mode is None:
             return examples[0]
 
+    def convert(self, data):
+        """Convert fetched data.
+
+        This method takes data fetched by :meth:`fetch` and
+        pre-process them before passing them to models.
+        The default behaviour is converting each column into an ndarray.
+        This behaviour can be overrided by :meth:`with_converter`.
```
Comment (beam2d) [inline chainer/dataset/tabular/tabular_dataset.py]:
```suggestion
        This behaviour can be overridden by :meth:`with_converter`.
```
Sorry, it's my bad :(

**Comment by beam2d** [comment]
LGTM.
Jenkins, test this please.

**Pull request merged**